src/FileFormats/LP/LP.jl

```diff
@@ -381,31 +383,31 @@ function _set_sense!(::Type{Val{:obj}}, model::Model, line)
     return MOI.set(model, MOI.ObjectiveSense(), _SENSE_ALIAS[lowercase(line)])
 end
 
-function _add_new_variable!(model::Model, name::String)
+function _add_new_variable!(model::Model, data_cache::CacheLPModel, name)
     var = MOI.add_variable(model)
     MOI.set(model, MOI.VariableName(), var, name)
+    data_cache.variables_in_model[name] = var
     return var
 end
 
-function _get_variable_from_name(model::Model, variable_name::String)
-    variables_inside_model = MOI.get(model, MOI.ListOfVariableIndices())
-    for var_inside_model in variables_inside_model
-        var_inside_model_name =
-            MOI.get(model, MOI.VariableName(), var_inside_model)
-        if var_inside_model_name == variable_name
-            return var_inside_model
-        end
+function _get_variable_from_name(
+    model::Model,
+    data_cache::CacheLPModel,
+    variable_name,
+)
+    var_inside_model = get(data_cache.variables_in_model, variable_name, "")
+    if var_inside_model != ""
```

Comment:
```suggestion
    if !isempty(var_inside_model)
```

or better yet make the default value `nothing` and then have `if var_inside_model !== nothing`

Reply:
for readability or performance?
current version is type stable. the `nothing` version relies on the small unions compiler optimization.